### ISSUE TITLE
Very high cpu usage when viewing large yarn.lock

### ISSUE THREAD

**Issue by mjbvz** [bartosz-antosik/vscode-spellright#229]
From https://github.com/Microsoft/vscode/issues/62973

- Extension Name: spellright
- Extension Version: 3.0.6
- OS Version: Darwin x64 18.2.0
- VSCode version: 1.29.0-insider

**repo**
1. Open a large yarn.lock file

**Bug**
notice large spike in cpu usage. I've attached a profile that shows operations in this extension taking a second or more: [CPU-20181112T223723.659Z.zip](https://github.com/bartosz-antosik/vscode-spellright/files/2573919/CPU-20181112T223723.659Z.zip)

**Comment by bartosz-antosik** [comment]
Could you please provide me with a yarn.lock file that causes the spikes? I do not know what yarn.lock is.

**Comment by bartosz-antosik** [comment]
I would like to know as what document type does yarn.lock open in VSCode and whether (if) this document type is defined by an extension, what extension is it?

**Comment by MaxMilton** [comment]
@bartosz-antosik `yarn.lock` files are a way to lock the dependencies version in a JS project, when using the yarn package manager. NPM has the same thing in `package-lock.json` files.

These files tend to be _huge_. On one of my projects it's over 10,000 lines long.

@mjbvz why are you trying to run spell check on a lockfile? It's not something you should ever manually edit.

In my user settings I ignore these kinds of files:

```
  "spellright.ignoreFiles": [
    "**/.gitignore",
    "**/.spellignore",
    "**/*.log",
    "**/package-lock.json",
    "**/tsconfig.json",
    "**/yarn.lock",
  ],
```

**Comment by mjbvz** [comment]
I'm not intentionally trying to spellcheck these files; it happens by default. 

Two problems here:

- `yarn.lock` should probably be excluded by default
- This extension should never cause VS Code to lock up, even for very large files.

**Comment by bartosz-antosik** [comment]
I agree that they should not lock or CPU pike Spell Right. Similar issue has been recently reported (#225) concerning large LaTeX documents and I agree there was some problem with large files. It has been investigated & I am not sure whether it works for other. Id seems to work for me.

Maybe you could please try to install this version from VSIX and see whether it reduces the delay in case of huge `yarn.lock` files:

https://drive.google.com/open?id=118oGg9HiGIec4BYtc4Th7IEEe4xW9xJb
